documentation/API-Testing-Guidelines.md

```diff
@@ -0,0 +1,41 @@
+# API Testing Guidelines
+
+## Table of Contents
+1. [Introduction](#introduction)
+2. [Test Cases contributions for APIs](#contribution)
+3. [Location of feature file](#location)
+4. [Best practices and recommendations](#recommendations)
+7. [References](#references)
+
+## Introduction <a name="introduction"></a>
+This document captures guidelines for the API testing in CAMARA project. These guidelines are applicable to every API to be worked out under the CAMARA initiative.
+
+## Test Cases contribution for APIs <a name="contribution"></a>
+Based on the decision taken in the commonalities working group and as documented in [API-Readiness-Checklist](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-Readiness-Checklist.md), a Gherkin feature file will be added to the main subproject repo and this will fullfil the minimum criteria of readiness w.r.t API test cases and documentation. If subprojects also intend to add test implementations, an aligned single implementation that is agreed amongst all provider implementers could be added to the main subproject repo. If no alignment is possible, each provider implementer will add the test implementation to their own repos.
+
+## Location of feature file <a name="location"></a>
+The feature file will reside under 
+```
+Subproject_Repository/code/Test_definitions/file_name.feature
+for e.g. https://github.com/camaraproject/QualityOnDemand/blob/main/code/Test_definitions/QoD_API_Test.feature
+```
+
+## Best practices and recommendations <a name="recommendations"></a>
+
+* One feature file per API is advisable so that all scenarios can be covered corresponding to the API & the corresponding resource.
```

Comment:
We should not impose one single .feature file per API, but one feature file per operation or API endpoint/resource. In general, it is not advisable to test several unrelated "API features" within the same .feature file. For closely related operations we may add scenarios in the same file testing crossed behaviour between several operations.

Reply:
@shilpa-padgaonkar following today Commonalities meeting I proposed to update this sentence but I'm not able to update the file. My proposal for line 25: 
Granularity of the feature file must be decided at the project level but it is recommended to:

- group in one file all scenarios for one given API capability (that can cover several endpoints). 
- provide several files when one CAMARA API (yaml) covers several independent functions that can be provided independently

Perhaps @jlurien  could provide a better wording. I think we're aligned.